### PR TITLE
Migrate remaining basic types to dc_bundle

### LINKED ISSUE #1453: Migrate remaining basic types to dc_bundle
This includes:

- toolkit_layout_style::Display
- toolkit_layout_style::Number
- toolkit_layout_style::Overflow
- toolkit_layout_style::LayoutSizing

### PR DESCRIPTION
**Pull request by timothyfroehlich**
# Based on PR #1458

## Full chain of PRs as of 2024-07-30

* PR #1459:
  `wb/froeht/1453-basic-types` ➔ `wb/froeht/migrate-meters`
* PR #1458:
  `wb/froeht/migrate-meters` ➔ `main`

Fixes #1453

### CODE REVIEW

File: crates/dc_bundle/src/legacy_definition/layout/grid.rs [hunk 1]
```diff
@@ -76,3 +76,39 @@ pub enum OverflowDirection {
...
+pub enum LayoutSizing {
+    #[default]
+    Fixed,
+    Hug,
+    Fill,
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum Display {
+    #[serde(rename = "flex")]
```
Comment (rylin8) [inline crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: Are these needed? I thought the `#serde(rename="xyz")` tag was to tell serde how to deserialize something, but this isn't used for deserialization I don't think.

Comment (timothyfroehlich) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: I don't know one way or another, but either way, they'll go away once we replace the types with the protobuf versions

Comment (rylin8) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: Also not sure if some of these belong in `grid`. `Display` I think can be for any layout type, and others like `LayoutSizing` are full autolayout.

Comment (timothyfroehlich) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: Yeah... it's fuzzy. Like I had the Display proto file that only had Display in it. I can move Display to `node_style`. What else should move?

Comment (rylin8) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: LayoutSizing, Overflow, OverflowDirection, and ItemSpacing should go to layout/layout_style or layout/positioning.

Comment (timothyfroehlich) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: I ran out of time to make these changes so I'll pick this up again when I'm back.

Comment (rylin8) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: ok, I'm fine with that being in a follow up pull request.

**Review by rylin8 — APPROVED** [review]

File: crates/dc_bundle/src/legacy_definition/layout/grid.rs (hunk 1 above)
Comment (timothyfroehlich) [reply crates/dc_bundle/src/legacy_definition/layout/grid.rs:104]: Changes now made

File: proto/definition/layout/grid.proto [hunk 2]
```diff
@@ -37,56 +37,17 @@ enum GridLayoutType {
...
-  }
-  oneof type {
-    int32 fixed = 1;
-    Auto auto = 2;
-  }
+
+
+
+// Configures whether a view is displayed using a Flexbox layout or not at all.
+enum Display {
```
Comment (rylin8) [inline proto/definition/layout/grid.proto:47]: Shouldn't this be in `node_style`?

Comment (timothyfroehlich) [reply proto/definition/layout/grid.proto:47]: Good catch

**Pull request merged**